src/client/mds_client.cpp

```diff
@@ -265,6 +267,10 @@ LIBCURVE_ERROR MDSClient::OpenFile(const std::string &filename,
 
         LIBCURVE_ERROR retcode = LIBCURVE_ERROR::FAILED;
         StatusCode stcode = response.statuscode();
+        if (StatusCode::kPermissionDeny == stcode &&
+            fi->context.openflags == CurveOpenFlags::CURVE_RDWR) {
+            return LIBCURVE_ERROR::PERMISSION_DENY;
+        }
```

Comment:
what's the purpose of this? I think if MDS returns an error, just return the error to the caller, and why test the open flags?